packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1113,22 +1147,44 @@ export default function buildKernel(
     }
 
     if (!kernelKeeper.isRunQueueEmpty()) {
-      return kernelKeeper.getNextMsg();
+      return kernelKeeper.getNextRunQueueMsg();
     }
     return undefined;
   }
 
+  function getNextAcceptanceMessage() {
+    if (!kernelKeeper.isAcceptanceQueueEmpty()) {
+      return kernelKeeper.getNextAcceptanceQueueMsg();
+    }
+    return undefined;
+  }
+
+  function maybeProcessNextMessage() {
```

Comment:
Not sure about the name.  I take the "maybe" here to mean "maybe there will be a message to process, maybe there won't, but if there is I will" rather than "maybe I'll process the next message, maybe I won't process it", but name hints at the latter.

Reply:
`processNextMessageIfAny` ?